### ISSUE TITLE
Migrate to RocksDB ByteBuffer APIs when Released

### ISSUE THREAD

**Issue by hraberg** [xtdb/xtdb#642]
Direct ByteBuffer support is finally coming to RocksDB JNI:
https://github.com/facebook/rocksdb/pull/2283

Once this is released we want to migrate our implementation to use it. We should then also kill off the JNR version of Rocks we have under `crux.rocksdb.kv.jnr`.

And as it will be moving in that direction we can then also potentially consolidate the LMDB side to use ByteBuffers only. The default version does, but the lmdbjava version could also be made do so, depends on if we want to keep LWJGL dependency or not.

Later we can potentially get rid of Agrona and JNR completely, but that's another card. Hashing would still use it, but JNR can work with Pointers and direct ByteBuffers if we want to keep it. A potential option is to get rid of the JNR hashing completely for simplicity, but it's there for performance reasons (~10%).

**Comment by hraberg** [comment]
Needless to say, it's worth sanity check the performance of the ByteBuffer API against our JNR implementation before removing it.

Note that Agrona buffers can be turned into ByteBuffers, so the Kv protocols would still talk Agrona, but we would ask for the ByteBuffer backing the UnsafeBuffer before calling down to Rocks/LMDB.

**Comment by hraberg** [comment]
This has now been released as part of RocksDB 6.8.0, will try to upgrade Crux to use this.

**Comment by hraberg** [comment]
The new ByteBuffer API cannot share buffers owned by Rocks, that's why you have to allocate one yourself and pass it in - the function returns number of read bytes, so if the buffer you passed was to small, you have to reallocate and go again - so by design it cannot work in the zero-copy fashion the JNR backend does.

Numbers, from `crux.kv-test/test-performance-off-heap`:

```
crux.kv.rocksdb/kv
"Elapsed time: 18405.202506 msecs"

crux.kv.rocksdb.jnr/kv
"Elapsed time: 12024.220405 msecs"

crux.kv.rocksdb.nio/kv
"Elapsed time: 14550.30613 msecs"
```
